packages/rlc-common/package.json

```diff
@@ -22,7 +22,7 @@
     "build": "rimraf --glob dist/* dist-esm/* types/* && tsc -p tsconfig.json && tsc -p tsconfig-cjs.json && node publishPackage.js",
     "build:cjs": "rimraf --glob dist/* dist-esm/* types/* && tsc -p tsconfig-cjs.json && node publishPackage.js",
     "build:esm": "rimraf --glob dist/* dist-esm/* types/* && tsc -p tsconfig.json && node publishPackage.js",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "mocha test/**/*.spec.ts"
```

Comment:
```suggestion
    "test": "cross-env TS_NODE_PROJECT=tsconfig.json mocha -r ts-node/register --experimental-specifier-resolution=node --experimental-modules=true  --timeout 4000 './test/unit/**/*.spec.ts'"
```
something like this ? 

Also, I am thinking about when we generate modular, we import the module as esm, otherwise, we will use cjs name  https://github.com/qiaozha/autorest.typescript/blob/5023ca95291e7f8b8374a5afa222d56cacd34aac/packages/rlc-common/src/helpers/nameConstructors.ts#L119 

Can loader as "ts-node/esm" cover cases in cjs ?

Reply:
I enabled the Unit Test for both cjs and esm in [commit](https://github.com/Azure/autorest.typescript/pull/2022/commits/7ec3a09020e824cfad643988fe101dac3ed36635). Please notice the suffix is not allowed in spec.ts file because cjs module can't find relevant dependencies. But it is okay when enabling option `experimental-specifier-resolution`.

![image](https://github.com/Azure/autorest.typescript/assets/9943211/eb7229b8-6649-405c-af7f-c2491eea5055)


Please also notice anothere detail for other packages in typespec-ts and autorest.typescript is we only enabled one kind of module resolution for their UT.

Reply:
I revert this change because I notice that our sources use the .js extenstion to find the dependency which will cause the cjs can't find module. I believe we could do this only based on the built code not directly on the sources using `ts-node/register`. Let's enhance this when necessary.

![image](https://github.com/Azure/autorest.typescript/assets/9943211/a3478e99-cd1d-4859-a1e8-8e6f51386b89)

Reply:
Do you find this issue exist in the both esm and cjs loader or just cjs loader ? if it's the later, would you mind change this way ?
`cross-env TS_NODE_PROJECT=tsconfig.json mocha -r ts-node/register --loader=ts-node/esm --experimental-specifier-resolution=node  test/**/*.spec.ts`

Reply:
`experimental-specifier-resolution` feature is for esm and you could refer the [link here](https://nodejs.cn/api/cli/experimental_specifier_resolution_mode.html).

Reply:
Got it, Thanks for the pointer.